### ISSUE TITLE
three problems with notepad->prompt

### ISSUE THREAD

**Issue by pryrt** [pryrt/Win32-Mechanize-NotepadPlusPlus#47]
1) in my experiments, it seems that notepad->prompt only works after notepad->messageBox has been run
If I comment out line 2, the script just freezes:
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
notepad->messageBox("abc");
print STDERR notepad->prompt("abc", "third", "default"), "\n";
```

2) If I use `<STDIN>`, it will freeze on ->prompt, even with ->messageBox first
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
notepad->messageBox("abc");
<STDIN>;
print STDERR notepad->prompt("abc", "third", "default"), "\n";
```

3) It's supposed to take three arguments (per docs), but only takes two -- the first goes to the title, the second to the default, nothing happens with the third; should be long-message, title, and default as the three args

**Comment by pryrt** [comment]
1) in my experiments, it seems that notepad->prompt only works after notepad->messageBox has been run
If I comment out line 2, the script just freezes:
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
notepad->messageBox("abc");
print STDERR notepad->prompt("abc", "third", "default"), "\n";
```

2) If I use `<STDIN>`, it will freeze on ->prompt, even 
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
notepad->messageBox("abc");
<STDIN>;
print STDERR notepad->prompt("abc", "third", "default"), "\n";
```

**Comment by pryrt** [comment]
the second sub-bug might be wrong.  Using that exact code, it now works right -- whether `<STDIN>` is line2 or line3.  Sometimes, however, the dialogs end up _behind_ the Notepad++ window.

But in my developing nppPasteSpecial, I think I was able to show that `$selection=<STDIN>;` would cause prompt to freeze, whereas $selection=1; would allow prompt to work.

**Comment by pryrt** [comment]
On (1), if I run
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
print STDERR "Input: "; my $txt = <STDIN>;
chomp $txt;
my $got = sprintf qq{got "%s"}, $txt;
print STDERR "$got\n";
print STDERR notepad->prompt('input title', $txt), "\n";
```
from _outside_ Notepad++ (an independent `cmd.exe` window), it properly prompts without freezing.

Note: all of these are using the same instance of Notepad++, without re-starting in between.  I am quoting my exact commands, with the `-I` arguments, because on the machine I'm currently on, I just use my development library, and don't have it installed to a standard `@INC` location.

If I run the same script _inside_ Notepad++ using NppExec, it freezes during the call to `prompt`.

If I run `cmd /c perl -I"c:/usr/local/share/GitHubSvn/Win32-Mechanize-NotepadPlusPlus/lib" "C:\usr\local\share\PassThru\perl\nppCommunity\nppPrompt.pl"` using **Run > Run**, it works! 

If I run **Plugins > NppConsole > ShowNppConsole** and run `perl -I"c:/usr/local/share/GitHubSvn/Win32-Mechanize-NotepadPlusPlus/lib" "C:\usr\local\share\PassThru\perl\nppCommunity\nppPrompt.pl"` from there, it works!

If I run @vinsworldcom's plsh command for "PerlScript", and run `print notepad->prompt('title', 'text'), "\n"`, it works!

So the only place I cannot get that to work is from NppExec!  I wonder what NppExec is highjacking that is causing the problem.

If Michael sees the @-mention and has a few spare minutes to confirm or contradict the quick experiments in this comment, that would be great.  (Mostly because this doesn't seem like the results I was getting yesterday.)

But, for now, I'll de-prioritize sub-bug (1) from this ticket, and focus first on getting the three-argument form to work.  I will probably come back and see if I can add debugging prints to get hints as to why

**Comment by pryrt** [comment]
Actually, as a simplified script for trying to replicate these results, don't even need the `<STDIN>` portion... just use
```
use Win32::Mechanize::NotepadPlusPlus qw/:main/;
print STDERR notepad->prompt('input title', 'default'), "\n";
```
This variant has all the same results as the above comment.

**Comment by vinsworldcom** [comment]
Can confirm - same results as you using the simple script.  All works fine, but when in NppExec console, just doing:

```
perl C:\users\VinsWorldcom\tmp\test.pl
```

"freezes" NppExec - no error, no prompt pop-up and no NppExec `======= READY =====` message indicating it's done.

If instead in the NppExec console, I do:

```
cmd /c perl C:\users\VinsWorldcom\tmp\test.pl
```

it works fine.

To be fair, most times when I launch and external from NppExec, I always preface with `cmd /c` so I'd probably never see this "bug".

Could be a NppExec limitation - try this, in NppExec console:

```
notepad
```

I get same behavior - no error, no READY returned and no notepad.exe launched,  Now try:

```
cmd /c notepad
```

It works!  NppExec may not be able to launch GUI apps without "forking" out to `cmd` first?

Cheers.

**Comment by pryrt** [comment]
> NppExec may not be able to launch GUI apps without "forking" out to cmd first?

That's almost certainly part of it.  

But when I include `messageBox` first in the same script, then both messageBox and prompt work.  

I confirm that prefixing the NppExec run with cmd/c will work for me, too.  Using `npp_run` instead of `cmd/c` also works.

So it's probably down to how NppExec calls things under the different syntaxes.

**Comment by pryrt** [comment]
* Close subitem 1 (freeze) => caused by NppExec not calling process correctly (prefix call with `cmd/c`) => not a bug
* Close subitem 2 (STDIN) => could not repeat
* Close subitem 3 (ignored third arg) => fixed in c3affe9

All subitems closed => close #47
